### PR TITLE
PHP 7.3

### PR DESCRIPTION
**Pull request by MGatner**
**Description**
* Updates the minimum PHP version to `7.3`
* Updates PHPUnit to version 9 and migrates config files
* Updates many version references in code and User Guide

**Checklist:**
- [X] Securely signed commits
- [X] Component(s) with PHPdocs
- [x] Unit testing, with >80% coverage
- [X] User guide updated
- [X] Conforms to style guide

### CODE REVIEW

File: admin/framework/phpunit.xml.dist [hunk 1]
```diff
@@ -1,60 +1,46 @@
...
-		backupGlobals="false"
-		colors="true"
-		convertErrorsToExceptions="true"
-		convertNoticesToExceptions="true"
-		convertWarningsToExceptions="true"
-		stopOnError="false"
-		stopOnFailure="false"
-		stopOnIncomplete="false"
-		stopOnSkipped="false">
+<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" bootstrap="system/Test/bootstrap.php" backupGlobals="false" colors="true" convertErrorsToExceptions="true" convertNoticesToExceptions="true" convertWarningsToExceptions="true" stopOnError="false" stopOnFailure="false" stopOnIncomplete="false" stopOnSkipped="false" xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.3/phpunit.xsd">
```
Comment (paulbalandan) [inline admin/framework/phpunit.xml.dist:2]: Can we make this as multiline for better readability?

File: admin/module/phpunit.xml.dist [hunk 2]
```diff
@@ -1,60 +1,46 @@
...
-		backupGlobals="false"
-		colors="true"
-		convertErrorsToExceptions="true"
-		convertNoticesToExceptions="true"
-		convertWarningsToExceptions="true"
-		stopOnError="false"
-		stopOnFailure="false"
-		stopOnIncomplete="false"
-		stopOnSkipped="false">
+<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" bootstrap="vendor/codeigniter4/framework/system/Test/bootstrap.php" backupGlobals="false" colors="true" convertErrorsToExceptions="true" convertNoticesToExceptions="true" convertWarningsToExceptions="true" stopOnError="false" stopOnFailure="false" stopOnIncomplete="false" stopOnSkipped="false" xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.3/phpunit.xsd">
```
Comment (paulbalandan) [inline admin/module/phpunit.xml.dist:2]: Also multiline

File: admin/starter/phpunit.xml.dist [hunk 3]
```diff
@@ -1,60 +1,46 @@
...
-		backupGlobals="false"
-		colors="true"
-		convertErrorsToExceptions="true"
-		convertNoticesToExceptions="true"
-		convertWarningsToExceptions="true"
-		stopOnError="false"
-		stopOnFailure="false"
-		stopOnIncomplete="false"
-		stopOnSkipped="false">
+<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" bootstrap="vendor/codeigniter4/framework/system/Test/bootstrap.php" backupGlobals="false" colors="true" convertErrorsToExceptions="true" convertNoticesToExceptions="true" convertWarningsToExceptions="true" stopOnError="false" stopOnFailure="false" stopOnIncomplete="false" stopOnSkipped="false" xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.3/phpunit.xsd">
```
Comment (paulbalandan) [inline admin/starter/phpunit.xml.dist:2]: Also this one

File: phpunit.xml.dist [hunk 4]
```diff
@@ -1,14 +1,24 @@
...
-		backupGlobals="false"
-		colors="true"
-		convertErrorsToExceptions="true"
-		convertNoticesToExceptions="true"
-		convertWarningsToExceptions="true"
-		stopOnError="false"
-		stopOnFailure="false"
-		stopOnIncomplete="false"
-		stopOnSkipped="false">
+<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" bootstrap="system/Test/bootstrap.php" backupGlobals="false" colors="true" convertErrorsToExceptions="true" convertNoticesToExceptions="true" convertWarningsToExceptions="true" stopOnError="false" stopOnFailure="false" stopOnIncomplete="false" stopOnSkipped="false" xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.3/phpunit.xsd">
```
Comment (paulbalandan) [inline phpunit.xml.dist:2]: And this one

**Comment by samsonasik** [comment]
`public/index.php` need to be updated as well.

**Review by paulbalandan — CHANGES_REQUESTED** [review]
index.php's min PHP should be changed too.

**Comment by paulbalandan** [comment]
Unrelated to this PR. I think when you rebased 4.1 to latest you mistakenly added `next` to the toctree. That was deleted before. This is causing errors in UG.

File: admin/framework/phpunit.xml.dist (hunk 1 above)
Comment (MGatner) [reply admin/framework/phpunit.xml.dist:2]: This is the output from PHPUnit's `--migrate-configuration`. I already converted it to tabs so I'm fine modifying it more - just noting where the differences come from.

**Review by paulbalandan — APPROVED** [review]

**Review by samsonasik — APPROVED** [review]

**Pull request merged**